src/main/java/io/cdap/plugin/format/DBTableRecordReader.java

```diff
@@ -159,13 +159,15 @@ public void close() throws IOException {
   }
 
   private String getQuery() {
-    if (split.getLowerClause() != null && split.getUpperClause() != null) {
-      // Now append the conditions associated with our split.
-      // Build the WHERE clauses associated with the data split.
-      String conditionClauses = "WHERE (( " + split.getLowerClause() + (" ) AND ( ") + split.getUpperClause() + " ))";
-      return dbConf.appendWhereClauseWithSplit("SELECT * FROM ", tableName.fullTableName(), conditionClauses);
+    String query = "SELECT * FROM " + tableName.fullTableName() + " ";
+    String whereClause = dbConf.getWhereClause();
+
+    if (whereClause != null && !whereClause.isEmpty()) {
+      query += split.getWhereClause() + " AND (" + whereClause.substring("where".length()) + ")";
```

Comment:
it's a little cleaner to make split.getWhereClause() not actually return 'where', so that you don't have to do a substring. Then this becomes:

```
if (...) {
  query += whereClause + " AND " + split.getWhereClause();
} else {
  query += "WHERE " + split.getWhereClause();
}

Reply:
done